buildozer/targets/osx.py

```diff
@@ -44,16 +45,19 @@ def download_kivy(self, cwd):
         else:
             if not exists(join(cwd, 'Kivy.dmg')):
                 self.logger.info('Downloading kivy...')
-                status_code = check_output((
-                    'curl', '-L', '--write-out', '%{http_code}',
-                    '-o', 'Kivy.dmg',
-                    f'https://kivy.org/downloads/{current_kivy_vers}/Kivy.dmg'),
-                    cwd=cwd)
-
-                if status_code == "404":
+                try:
+                    buildops.download(
+                        f'https://kivy.org/downloads/'
+                        f'{current_kivy_vers}/Kivy.dmg',
+                        'Kivy.dmg',
+                        cwd=cwd
+                    )
+                except urllib.Error:
```

Comment:
Does `urllib.Error` exists?

These: https://github.com/python/cpython/blob/3.11/Lib/urllib/error.py are the valid errors that `urllib` can raise.

Reply:
Will fix. This one annoys me because I deliberately set up an prototype to see what errors actually got raised, and then I managed to write it down wrong. And static code analysis didn't save me. :-(